cpp/src/slice2cs/Gen.cpp

```diff
@@ -103,29 +103,65 @@ namespace
     }
 
     /// Returns a C# formatted link to the provided Slice identifier.
-    /// TODO: this is temporary and will be replaced when we add 'cs:identifier' support.
-    string csLinkFormatter(const string& rawLink, const ContainedPtr&, const SyntaxTreeBasePtr&)
+    string csLinkFormatter(const string& rawLink, const ContainedPtr& source, const SyntaxTreeBasePtr& target)
     {
-        string result = "<see cref=\"";
+        ostringstream result;
+        result << "<see ";
 
-        auto hashPos = rawLink.find('#');
-        if (hashPos != string::npos)
+        if (target)
         {
-            if (hashPos != 0)
+            if (auto builtinTarget = dynamic_pointer_cast<Builtin>(target))
             {
-                result += rawLink.substr(0, hashPos);
-                result += ".";
+                string typeS = CsGenerator::typeToString(builtinTarget, "");
+                if (typeS.find("Ice") == std::string::npos)
+                {
+                    result << "langword=\"" << typeS << "\"";
+                }
+                else
+                {
+                    // Remove trailing '?':
+                    typeS.pop_back();
+                    result << "cref=\"" << typeS << "\"";
+                }
+            }
+            else
+            {
+                // TODO: remove once mappedScope() is fixed.
+                string sourceScope = source->mappedScope(".").substr(1);
+                sourceScope.pop_back();
+
+                if (auto operationTarget = dynamic_pointer_cast<Operation>(target))
+                {
+                    // link to the method on the proxy interface
+                    result << "cref=\"" << CsGenerator::getUnqualified(operationTarget->interface(), sourceScope)
+                           << "Prx." << operationTarget->mappedName() << "\"";
+                }
+                else if (auto interfaceTarget = dynamic_pointer_cast<InterfaceDecl>(target))
+                {
+                    // link to the proxy interface
+                    result << "cref=\"" << CsGenerator::getUnqualified(interfaceTarget, sourceScope) << "Prx\"";
+                }
+                else
+                {
+                    result << "cref=\""
+                           << CsGenerator::getUnqualified(dynamic_pointer_cast<Contained>(target), sourceScope) << "\"";
+                }
             }
-            result += rawLink.substr(hashPos + 1);
         }
         else
         {
-            result += rawLink;
+            // Replace "::"" by "." in the raw link. This is for the situation where the user passes a Slice type
+            // reference but (a) the source Slice file does not include this type and (b) there is no cs:identifier or
+            // other identifier renaming.
```

Comment:
Do we need to deal with `#` of Slice links or that is already done at this point?

Reply:
I don't think it is or should be since it's called "rawLink".

Reply:
> Do we need to deal with # of Slice links or that is already done at this point?

The link will still have any `#`s in it. No processing has been done to it by this point. It's all up to this function.

> A raw link might be invalid for the cref value.

I think we should make a best effort attempt to map it.
Which should be as simple as adding `std::replace(s.begin(), s.end(), '#', '.');`

Reply:
Do we also emit a warning when target was not found?

Reply:
Fixed as suggested.